### PR TITLE
Push to quay.io as well

### PR DESCRIPTION
**Pull request by terbepetra**
## Describe your changes
Modified `/metadata-action` and added Quay.io login in our `build-and-push-image` workflow so We will push images to Quay.io registry as well.

## How best to test these changes
The modified check ran for this PR.

### CODE REVIEW

File: .github/workflows/push-docker-image.yaml [hunk 1]
```diff
@@ -70,3 +70,17 @@ jobs:
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Push To quay.io
+        id: push-to-quay
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          context: .
```
Comment (rbarabas) [inline .github/workflows/push-docker-image.yaml:78]: Warning: Unexpected input(s) 'context', 'file', 'platforms', 'push', 'labels', valid inputs are ['image', 'tags', 'registry', 'username', 'password', 'tls-verify', 'digestfile', 'extra-args']

The keyword `context` has no meaning here. 

This particular Github Action only supports `['image', 'tags', 'registry', 'username', 'password', 'tls-verify', 'digestfile', 'extra-args']`

Comment (rbarabas) [reply .github/workflows/push-docker-image.yaml:78]: Did you mean to use `docker/build-push-action@v1`?

**Comment by rbarabas** [comment]
To conclude what we discussed offline: review how the workflow works and try to interpret it. 

I'd building up a publishing workflow using https://github.com/docker/build-push-action/tree/releases/v1#registry and https://github.com/docker/login-action/tree/releases/v1#quayio. If you need help, LMK!

**Comment by terbepetra** [comment]
TODO: tagging

**Comment by terbepetra** [comment]
`docker/metadata-action@v4` was added for testing reasons

**Review by rbarabas — CHANGES_REQUESTED** [review]
Could we do a single build with multiple pushes? 

Something along the lines of what's outlined in https://docs.docker.com/build/ci/github-actions/push-multi-registries/. The `cache-from` and `cache-to` parameters and their version specific variants like [cache-froms in v1](https://github.com/docker/build-push-action/tree/releases/v1#cache_froms) look promising.

File: .github/workflows/push-docker-image.yaml [hunk 2]
```diff
@@ -55,12 +64,25 @@ jobs:
...
+        run: |
+          echo "${{ env.DOCKER_METADATA_OUTPUT_VERSION }}"
+
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
 
+      # - name: Build and push Quay.io images
```
Comment (rbarabas) [inline .github/workflows/push-docker-image.yaml:77]: Could you remove the commented step here?

File: .github/workflows/push-docker-image.yaml [hunk 3]
```diff
@@ -55,12 +64,25 @@ jobs:
             type=semver,pattern={{major}}.{{minor}}
             type=raw,value=latest,enable=${{ github.event_name == 'release' }}
 
+      - name: Debug tags - print
```
Comment (rbarabas) [inline .github/workflows/push-docker-image.yaml:67]: We can probably remove this in the final product.

File: docker-quay/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2022-2023 Tigris Data, Inc.
```
Comment (rbarabas) [inline docker-quay/Dockerfile:1]: This file is no longer necessary, correct?

Comment (terbepetra) [reply docker-quay/Dockerfile:1]: That is right. It was only needed for Quay's `build-push-action@v1` action

**Pull request merged**